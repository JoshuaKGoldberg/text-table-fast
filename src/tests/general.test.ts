import { describe, expect, test } from "vitest";

import { textTable } from "../index.js";

describe("general", () => {
	test.each([
		[[["main", "0123456789abcdef"]]],
		[
			[
				["main", "0123456789abcdef"],
				["staging", "fedcba9876543210"],
			],
		],
		[
			[
				["master", "0123456789abcdef"],
				["staging", "fedcba9876543210"],
			],
		],
	])("%j", (rows) => {
		expect("\n" + textTable(rows) + "\n").toMatchSnapshot();
	});
});
