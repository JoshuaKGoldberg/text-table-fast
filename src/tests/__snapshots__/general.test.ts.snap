// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`general > [["main","0123456789abcdef"],["staging","fedcba9876543210"]] 1`] = `
"
main     0123456789abcdef
staging  fedcba9876543210
"
`;

exports[`general > [["main","0123456789abcdef"]] 1`] = `
"
main  0123456789abcdef
"
`;

exports[`general > [["master","0123456789abcdef"],["staging","fedcba9876543210"]] 1`] = `
"
master   0123456789abcdef
staging  fedcba9876543210
"
`;
